operator/pkg/reconciliation/secrets.go

```diff
@@ -227,19 +227,23 @@ func (rc *ReconciliationContext) retrieveInternodeCredentialSecretOrCreateDefaul
 func validateCassandraUserSecretContent(dc *api.CassandraDatacenter, secret *corev1.Secret) []error {
 	var errs []error
 
-	namespacedName := types.NamespacedName{
-		Name:      secret.ObjectMeta.Name,
-		Namespace: secret.ObjectMeta.Namespace,
-	}
-	errorPrefix := fmt.Sprintf("Validation failed for user secret: %s", namespacedName.String())
-
-	for _, key := range []string{"username", "password"} {
-		value, ok := secret.Data[key]
-		if !ok {
-			errs = append(errs, fmt.Errorf("%s Missing key: %s", errorPrefix, key))
-		} else if !utf8.Valid(value) {
-			errs = append(errs, fmt.Errorf("%s Key did not have valid utf8 value: %s", errorPrefix, key))
+	if secret != nil {
```

Comment:
nice find! I'm curious how we got here... did we check if one secret exists and then assume multiple existed?

Reply:
I don't see any being checked, there was a parameter in the test to check it, but commit 9b376e8be93976a0a344bcda2e417aa90dd9758f removed those tests (and broke the functionality).